path_login.go

```diff
@@ -190,15 +187,12 @@ func (b *kubeAuthBackend) aliasLookahead(ctx context.Context, req *logical.Reque
 		return logical.ErrorResponse(fmt.Sprintf("invalid role name %q", roleName)), nil
 	}
 
-	// Parse into JWT
-	parsedJWT, err := jws.ParseJWT([]byte(jwtStr))
+	config, err := b.loadConfig(ctx, req.Storage)
 	if err != nil {
 		return nil, err
 	}
 
-	// Decode claims into a service account object
-	sa := &serviceAccount{}
-	err = mapstructure.Decode(parsedJWT.Claims(), sa)
+	sa, err := b.parseAndValidateJWT(jwtStr, role, config)
```

Comment:
Is it worth leaving a comment behind to ensure the intent is clear and help prevent similar bugs in future patches? It seems like an easy mistake to make.

Reply:
Fixed in. https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/114/commits/aec50977ce5b5f1aab03c38c4153abec583105fe